modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/js/Comments.js

```diff
@@ -437,4 +437,80 @@ export default function ({
 
 		editorInstance.dispose();
 	};
+
+	window[`${randomNamespace}onMessagePosted`] = function (
+		response,
+		refreshPage
+	) {
+		Liferay.onceAfter(`${portletDisplayId}:portletRefreshed`, () => {
+			const randomNamespaceNodes = document.querySelectorAll(
+				`input[id^="${namespace}"][id$="randomNamespace"]`
+			);
+
+			Array.prototype.forEach.call(
+				randomNamespaceNodes,
+				(node, index) => {
+					const randomId = node.value;
+
+					if (index === 0) {
+						window[`${randomNamespace}showStatusMessage`]({
+							id: randomId,
+							message: Liferay.Language.get(
+								'your-request-completed-successfully'
+							),
+							type: 'success',
+						});
+					}
+
+					const currentMessageSelector = `${randomId}message_${response.commentId}`;
+
+					const targetNode = document.getElementById(
+						currentMessageSelector
+					);
+
+					if (targetNode) {
+						location.hash = '#' + currentMessageSelector;
+					}
+				}
+			);
+		});
+
+		const className = Util.getFormElement(form, 'className').value;
+		const classPK = Util.getFormElement(form, 'classPK').value;
+
+		if (response.commentId) {
+			const messageTextNode = document.querySelector(
+				`input[name^="${namespace}${randomNamespace}body"]`
+			);
+
+			if (messageTextNode) {
+				Liferay.fire('messagePosted', {
+					className,
+					classPK,
+					commentId: response.commentId,
+					text: messageTextNode.value,
+				});
+			}
+		}
+
+		if (refreshPage) {
+			window.location.reload();
+		}
+		else {
+			const portletNodeId = `p_p_id_${portletDisplayId}_`;
+
+			const portletNode = document.getElementById(portletNodeId);
```

Comment:
This is not correct. Previously we had `var portletNode = A.one(portletNodeId)` which returned an object with `refreshURLData` that now we are missing. Asked in [FI channel](https://liferay.slack.com/archives/CNBG06JS3/p1624619688234200?thread_ts=1624619616.234100&cid=CNBG06JS3) for help.